### PR TITLE
Update RadMon's build script to work with the CMake refactor

### LINKED ISSUE #389: Update RadMon build script
With the CMake refactor the RadMon's build script no longer works and will need to be updated.

### PR DESCRIPTION
**Pull request by EdwardSafford-NOAA**
Update build script and light code clean up to remove commented out code and unnecessary debug output.

### CODE REVIEW

**Comment by MichaelLueken** [comment]
@EdwardSafford-NOAA Would you like anyone to review the changes, or are these changes ready to be merged?  If they are ready now, I will be able to merge them tomorrow after the current work out for review has been reviewed.  Otherwise, I will assign whoever you would like to have review these changes.  Also, it is noted that there are two commits associated with this work.  To bring this down to the single commit, please do the following:

1) git clone --recursive git@github.com:EdwardSafford-NOAA/GSI.git update
2) cd update
3) git reset --hard HEAD~1
4) git reset --hard HEAD~1
5) git remote add upstream https://github.com/NOAA-EMC/GSI.git
6) git remote update
7) git rebase upstream/develop
8) git checkout RadMon_389
9) git checkout develop
10) git merge --squash RadMon_389
11) git reset HEAD src/ncdiag/cmake/ncdiag_compiler_flags_Intel_Fortran.cmake util/cmake/gsiutils_compiler_flags_Intel_Fortran.cmake
12) git checkout -- src/ncdiag/cmake/ncdiag_compiler_flags_Intel_Fortran.cmake util/cmake/gsiutils_compiler_flags_Intel_Fortran.cmake
13) git commit -m `"Github issue #389.  Update RadMon's package level build to work with refactored CMake.  Fixes #389."`
14) git push origin develop --force

**Comment by EdwardSafford-NOAA** [comment]
@MichaelLueken-NOAA you or @aerorahul should double check my modification of the gsi's CMakeList file.  It works for both GSI level and RadMon level builds now but I'm not a CMake expert, so I may have missed something.

I'm sorry I didn't catch the two commits.  I'm really unsure how I managed to do that.  I'll fix it now.

File: util/CMakeLists.txt [hunk 1]
```diff
@@ -153,6 +155,10 @@ if(BUILD_UTIL_MON)
   add_subdirectory(Radiance_Monitor)
 endif()
 
+if(BUILD_UTIL_RAD)
+  add_subdirectory(Radiance_Monitor)
+endif()
```
Comment (aerorahul) [inline util/CMakeLists.txt:160]: `Radiance_Monitor` is being built with the conditional `BUILD_UTIL_MON`, and will conflict with it on L155.

@EdwardSafford-NOAA 
Is there a special need for building just the `Radiance_Monitor`?

If so, I suggest, deprecating `BUILD_UTIL_MON` and replace them with 3 separate options; one each for Conventional, Ozone and Radiance as `BUILD_UTIL_CONMON`, `BUILD_UTIL_OZNMON` and `BUILD_UTIL_RADMON` respectively.

File: util/Radiance_Monitor/build_RadMon_cmake.sh [hunk 2]
```diff
@@ -3,139 +3,75 @@
...
+# Get the root of the cloned GSI directory
+readonly GSI_ROOT=$(cd "$(dirname "$(readlink -f -n "${BASH_SOURCE[0]}" )" )/../.." && pwd -P)
+
+echo "GSI_ROOT = ${GSI_ROOT}"
+
+#-------------------------------------------------------
+# Modify CoMakeLists.txt to turn off GSI and ENKF builds
+#-------------------------------------------------------
+cml=${GSI_ROOT}/CMakeLists.txt
+cp ${cml} ${cml}.orig
```
Comment (aerorahul) [inline util/Radiance_Monitor/build_RadMon_cmake.sh:22]: Do not modify `CMakeLists.txt` or move them. Please use `cmake` options to turn `ON|OFF`

File: util/Radiance_Monitor/build_RadMon_cmake.sh [hunk 3]
```diff
@@ -3,139 +3,75 @@
...
 #
-#  This script builds all of the executables in the 
-#  nwprod/radmon_shared/exec, data_extract/exec, and image_gen/exec
-#  subdirectories.  
-#
-#  The operational RadMon executables (in nwprod/radmon_shared/exec) 
-#  may also be built as part of the whole GSI package.  To do this 
-#  ensure BUILD_UTIL=ON when running cmake or use the 
-#  ProdGSI/ush/build_all_cmake.sh script.
+#  This script builds all of the RadMon executables and moves them
```
Comment (aerorahul) [inline util/Radiance_Monitor/build_RadMon_cmake.sh:6]: It is difficult to understand what is going on in this script.
IMO, there should not be a `build_RadMon_cmake.sh` script.
One should be able to build everything (and anything) from the top-level `CMake`.

@EdwardSafford-NOAA Can you please explain, what are the modifications being doing to the various files and why?

What is the desired end result?

**Review by aerorahul — CHANGES_REQUESTED** [review]
I have only reviewed the changes to the `CMakeLists.txt` and `build_RadMon_cmake.sh`.
I am unclear why there needs to exist a `build_RadMon_cmake.sh` and the changes in the `CMakeLists.txt` will cause conflicts.

File: util/Radiance_Monitor/build_RadMon_cmake.sh (hunk 3 above)
Comment (EdwardSafford-NOAA) [reply util/Radiance_Monitor/build_RadMon_cmake.sh:6]: @aerorahul The RadMon level build script is primarily a convenience for builds during development and package installation.  Strictly speaking it's not necessary and building everything from the top level will suffice.  There always has been a package level build capability, but that is a legacy of early building mechanisms, and has influenced the way I've viewed this. 

I'm good with removing the `build_RadMon_cmake.sh` script. I'll close this PR and rework these changes accordingly.

**Pull request closed without merging**